### PR TITLE
feat: application version in footer

### LINKED ISSUE #459: Add release notes to aescan.io
Hello,

It would be great to have the release notes page, like they are in GitHub available e.g. new changes, version aso.

Thanks,
David

### PR DESCRIPTION
**Pull request by lukeromanowicz**
## Description
Resolves #459

## Demo
image

'develop' will have `-dev` suffix, while prod will have no suffixes at all. 

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

### CODE REVIEW

**Review by CharisSiarampalis — APPROVED** [review]

**Review by janmichek — APPROVED** [review]
Crazy cool, I like this feature a lot!


I would prefer to navigate to the particular version in the changelog (to anchor), because the latest one does not need to be deployed one yet
EG: https://github.com/aeternity/ae_mdw/blob/master/CHANGELOG.md#1570-2023-08-03 but I guess it's not possible given the link structure

File: nuxt.config.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ export default defineNuxtConfig({
     apiHost: 'https://plausible.dev.service.aepps.com',
     trackLocalhost: false,
   },
+  appConfig: {
+    APP_VERSION: process.env.APP_VERSION || 'UNKNOWN',
```
Comment (michele-franchi) [inline nuxt.config.ts:24]: Can't we simply hide that if it's empty? So the `|| ' UNKNOWN'` is not needed

File: src/components/TheFooter.vue [hunk 2]
```diff
@@ -40,12 +40,22 @@
         </div>
         <div class="footer__column">
           <div class="footer__network">
-            <div class="footer__version">
+            <app-link
```
Comment (michele-franchi) [inline src/components/TheFooter.vue:43]: In my opinion I would just add the link to the version tag and not the whole text, looks a bit better to me.

File: src/components/TheFooter.vue [hunk 3]
```diff
@@ -77,6 +87,11 @@ import { useStatus } from '@/stores/status'
 
 const { middlewareVersion, nodeVersion } = storeToRefs(useStatus())
 const { MIDDLEWARE_URL } = useRuntimeConfig().public
+const { APP_VERSION } = useAppConfig()
+
+const APP_CHANGELOG_URL = 'https://github.com/aeternity/aescan/blob/main/CHANGELOG.md'
```
Comment (michele-franchi) [inline src/components/TheFooter.vue:92]:
```suggestion
const APP_RELEASES_URL = 'https://github.com/aeternity/aescan/releases'
```

I would use the releases link here

File: src/components/TheFooter.vue [hunk 4]
```diff
@@ -77,6 +87,11 @@ import { useStatus } from '@/stores/status'
...
+const NODE_CHANGELOG_URL = 'https://github.com/aeternity/aeternity/releases'
+const MDW_CHANGELOG_URL = 'https://github.com/aeternity/ae_mdw/blob/master/CHANGELOG.md'
```
Comment (michele-franchi) [inline src/components/TheFooter.vue:94]:
```suggestion
const MDW_RELEASES_URL = 'https://github.com/aeternity/ae_mdw/releases'
```

I would use the releases link here

File: src/components/TheFooter.vue [hunk 5]
```diff
@@ -77,6 +87,11 @@ import { useStatus } from '@/stores/status'
 
 const { middlewareVersion, nodeVersion } = storeToRefs(useStatus())
 const { MIDDLEWARE_URL } = useRuntimeConfig().public
+const { APP_VERSION } = useAppConfig()
+
+const APP_CHANGELOG_URL = 'https://github.com/aeternity/aescan/blob/main/CHANGELOG.md'
+const NODE_CHANGELOG_URL = 'https://github.com/aeternity/aeternity/releases'
```
Comment (michele-franchi) [inline src/components/TheFooter.vue:93]:
```suggestion
const NODE_RELEASES_URL = 'https://github.com/aeternity/aeternity/releases'
```

**Review by michele-franchi — CHANGES_REQUESTED** [review]
I added some personal suggestions but in general it looks good!

File: nuxt.config.ts (hunk 1 above)
Comment (lukeromanowicz) [reply nuxt.config.ts:24]: Great idea, didn't think of that :sweat_smile:

File: src/components/TheFooter.vue (hunk 2 above)
Comment (lukeromanowicz) [reply src/components/TheFooter.vue:43]: This won't work on on-production. Is that ok?

Comment (michele-franchi) [reply src/components/TheFooter.vue:43]: Why wouldn't it work? It's just about to not add the link to the service name. I don't get the difference

Comment (lukeromanowicz) [reply src/components/TheFooter.vue:43]: Sorry, minsunderstading. Of course it will :) I'll do that

**Comment by lukeromanowicz** [comment]
@michele-franchi great ideas, I believe everything is fixed.

@janmichek I love that feature but I decided not to implement linking directly to a specific tag because:
1. it will not work for aescan outside production because we create releases only for released code. That means on develop and previews aescan link will navigate to github error page
2. if middleware or core team removes some release or tag the code differently than the release itself (intentionally or not) it will again navigate users to github error page. I think it's safer to link users to releases page which is always working.

I hope it makes sense

**Review by michele-franchi — APPROVED** [review]
Good job 👏

**Pull request merged**